src/plugin/memory-provider.ts

```diff
@@ -143,24 +143,34 @@ export class MemoryProvider {
         return vscode.debug.activeDebugSession;
     }
 
-    public async readMemory(args: DebugProtocol.ReadMemoryArguments): Promise<ReadMemoryResult> {
-        return sendRequest(this.assertCapability('supportsReadMemoryRequest', 'read memory'), 'readMemory', args);
+    public async readMemory(args: [DebugProtocol.ReadMemoryArguments, Context?]): Promise<ReadMemoryResult> {
+        const session = this.assertCapability('supportsReadMemoryRequest', 'read memory');
+        const handler = this.adapterRegistry?.getHandlerForSession(session.type);
+        if (handler?.readMemory) { return handler.readMemory(session, args[0], args[1]); }
+        return sendRequest(session, 'readMemory', args[0]);
```

Comment:
Destructuring might make this clearer:
In `messaging.ts`

```typescript
type ReadMemoryWithContext = [ReadMemoryArguments, Context?];
```

```suggestion
    public async readMemory([readMemoryArgs, context]: ReadMemoryWithContext): Promise<ReadMemoryResult> {
        const session = this.assertCapability('supportsReadMemoryRequest', 'read memory');
        const handler = this.adapterRegistry?.getHandlerForSession(session.type);
        if (handler?.readMemory) { return handler.readMemory(session, readMemoryArgs, context); }
        return sendRequest(session, 'readMemory', readMemoryArgs);
```